### PR TITLE
Recipe app

### PR DESCRIPTION
**Pull request by enis-memic**
### In this PR @dheerajsachdeva and @enis-memic did the following:
- A login page.
- A registration page.
- A food list (with all CRUD implementation, except for 'update').
- A recipes list (with all CRUD implementation, except for 'update').
- A general shopping list view (all missing food for all your recipes and total price).
- A list of all public recipes from other users with their names and total prices.

### Project requirements

- You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
- Login page and registration page:
    - Should be built with Devise.
- Food list (**only for a group that has less than 3 members**):
    - Should display a list of food added by the logged-in user as in the wireframe (display also quantity of a given food).
    - Should lead to a form that allows users to add new food.
- Recipes list:
    - Should display a list of recipes created by the logged-in user as in the wireframe.
    - Should lead to recipe details.
    - If the user is the owner of the recipe, should allow the user to delete it.
- Public recipe list:
    - Should display a list of all public recipes ordered by newest as in the wireframe.
    - Should lead to recipe details.
- Recipe details:
    - Should display a toggle button that allows for a recipe to be made public or private.
    - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
    - If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- General shopping list view (**only for a group that has less than 3 members**):
    - Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
    - Should count the total food items and total price of the missing food.
- Make sure there are no N+1 queries happening.
- Create a navigation menu that allows users to open all of the pages you created.
- Write unit and integration tests.

### CODE REVIEW

File: app/views/recipes/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    
+<% @recipes.each do |recipe| %> <div class="card mb-3">
+ <div class="card-header d-flex justify-content-between align-items-center">
+ <%= link_to recipe.name, recipe_path(recipe.id) %>
+ <% if current_user.present? && can?(:destroy, recipe) %>
+ <%= button_to "Delete", recipe_path(recipe.id), method: :delete, data: { confirm: "Are you sure?" }, class: 'btn btn-danger btn-sm' %> 
+ <% end %> 
+ </div><div class="card-body">
+ <p class="card-text"><%= recipe.description %></p>    </div>   </div>  <% end %>
+</div>
```
Comment (franclobo) [inline app/views/recipes/index.html.erb]: - [x] Kindly correct the UI of the recipe list to be similar to the wireframe 👇🏽 

| Wireframe | Your design |
|---|---|
| image | image |

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @enis-memic and @dheerajsachdeva, 

# ♻️ REQUIRED CHANGES ♻️ 

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Good UI ✔️ 
- All tests passed ✔️ 


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/views/recipes/index.html.erb (hunk 1 above)
Comment (enis-memic) [reply app/views/recipes/index.html.erb]: Thanks for that suggestion.

**Comment by HAMMAS-SALEEM** [comment]
# Status: Approved :green_circle: :1st_place_medal: :100:
success

### Hey @enis-memic, @dheerajsachdeva 👋

Great work on implementing the requested changes. Your project is complete. :blush:
I just wanted to let you know that there is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following Requirements
✔️ Your app is working as expected 👍
✔️ Great work on following the wireframe 🥇

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval. Still, we highly recommend to students that they should take these changes into consideration._

Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**